staging/src/k8s.io/cli-runtime/pkg/resource/helper.go

```diff
@@ -190,7 +190,7 @@ func (m *Helper) DeleteWithOptions(namespace, name string, options *metav1.Delet
 	if options == nil {
 		options = &metav1.DeleteOptions{}
 	}
-	if m.ServerDryRun {
+	if m.ServerDryRun && len(options.DryRun) == 0 {
```

Comment:
I was wondering why these needed to change and found your commit message:
> helper currently has mandatory fields. this change makes it so it only overrides the request options if the request options did not request this

So, what this changeset does is change which of the helper struct's values and the options passed in takes precedence. I looked around to see if any existing callers seem to have expectations around that. Here's an example:

```go
	deleteResponse, err := resource.
		NewHelper(info.Client, info.Mapping).
		DryRun(o.DryRunStrategy == cmdutil.DryRunServer).
		DeleteWithOptions(info.Namespace, info.Name, deleteOptions)
```

In this case, I think the caller is expecting its explicit `.DryRun` to take precedence. Would you agree? I assume the place we needed this change for is our `.Patch` calls. Could we instead do the `helper.WithFieldManager(fieldManager).Patch()` pattern?

Reply:
thanks for calling this out. Unfortunately the `WithFieldManager` function is not functional as it appears, it imperatively sets it, so would need to reset the field manager back after the operation. 

You are right it is risky to change the semantics out from under other usages. Open to suggestions. My main goal was to reuse the existing `Apply` helper to supply a field manager to `Patch` without creating too much code smell.

Reply:
I've removed the problematic commit, and am now copying the helper object to set the field manager at point-of-use.